tests/e2e/tidbcluster/serial.go

```diff
@@ -589,6 +589,245 @@ var _ = ginkgo.Describe("[Serial]", func() {
+				}
+				log.Logf("confirm tikv pods haven't been changed this time")
+
+				return false, nil
+			})
+			framework.ExpectEqual(err, wait.ErrWaitTimeout, "expect tikv haven't been changed for 5 minutes")
+			log.Logf("Upgrade TidbCluster 1 with new version, but not rolling update.")
+
+			ginkgo.By("Set label version=old to TidbCluster 1, check that TiKV of TidbCluster 1 has been rolling updated")
+			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+				tc.Labels = map[string]string{"version": "old"}
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to update TidbCluster 1 to set label version=old")
+
+			err = wait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+				// confirm the tidb has been changed
+				changed, err := utilpod.PodsAreChanged(c, tikvPods)()
+				if err != nil {
+					log.Logf("ERROR: meet error during verify tikv pods, err:%v", err)
+					return true, nil
+				}
+				if changed {
+					return true, nil
+				}
+				log.Logf("confirm tikv pods haven't been changed this time")
+
+				return false, nil
+			})
+			framework.ExpectNoError(err, "expect tikv pods have been changed for 5 minutes")
+			log.Logf("Upgrade TidbCluster 1 with label version=old and tikv pods have been rolling updated.")
+
+			ginkgo.By("Deploy TidbCluster 2 with label version=new")
+			tc2Name := "tidbcluster2"
+			tc2 := fixture.GetTidbCluster(ns, tc2Name, utilimage.TiDBV4Prev)
+			tc2.Spec.PD.Replicas = 3
+			tc2.Spec.TiKV.Replicas = 3
+			tc2.Spec.TiDB.Replicas = 1
+			tc2.Labels = map[string]string{"version": "new"}
+
+			err = genericCli.Create(context.TODO(), tc2)
+			framework.ExpectNoError(err, "Expected TiDB cluster created")
+			log.Logf("Finished deploying TidbCluster 2 with label version=new.")
+
+			ginkgo.By("Wait for 5 minutes and check that no tikv Pod be created.")
+			getPods = func(ls string) ([]v1.Pod, error) {
+				listOptions := metav1.ListOptions{
+					LabelSelector: ls,
+				}
+				podList, err := c.CoreV1().Pods(ns).List(listOptions)
+				if err != nil {
+					return nil, err
+				}
+				return podList.Items, nil
+			}
+
+			err = wait.Poll(5*time.Second, 5*time.Minute, func() (created bool, err error) {
+				// confirm the Tidbcluster 2 tidb Pod don't be created
+				tikvPods, err = getPods(labels.SelectorFromSet(label.New().Instance(tc2Name).TiKV().Labels()).String())
+				if err != nil {
+					log.Logf("ERROR: meet error during get tikv pods, err:%v", err)
+					return true, nil
+				}
+				if len(tikvPods) != 0 {
+					return true, nil
+				}
+				log.Logf("confirm tikv pods have be created this time")
+				return false, nil
+			})
+			framework.ExpectEqual(err, wait.ErrWaitTimeout, "confirm Tidbcluster 2 no tikv pod be created")
+			log.Logf("confirm Tidbcluster 2 no tikv pod be created")
+
+			ginkgo.By("Deploy TiDB Operator 2 with --selector=version=new")
+			operatorVersion = "v1.1.11"
+			ocfg2 := &tests.OperatorConfig{
+				Namespace:       ns + "-canary-deploy",
```

Comment:
This can use the same namespace if deploy the current Operator so that it can cover the case in #3819.

Reply:
For now, we can't delete operator1 cause the scheduler doesn't support canary deploy. So I use deploy two operators in different namespaces and turnoff AppendReleaseSuffix.